### PR TITLE
MODSOURCE-239: Upgrade to RAML Module Builder 32.x

### PR DESCRIPTION
**Pull request by VRohach**
## Purpose
The main purpose of this PR is to upgrade the module to the new version of RMB 32.x and Vertx 4.
Moreover, there are some library updates for this change. 
In that libraries, where API was changed, implementation was improved.

## Approach
All changes which are needed to support RMB 32.x and Vertx 4.x were applied.

## Learning
More info: https://issues.folio.org/browse/MODSOURCE-239

### CODE REVIEW

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java [hunk 1]
```diff
@@ -707,7 +716,7 @@ private ReactiveClassicGenericQueryExecutor getQueryExecutor(String tenantId) {
     return postgresClientFactory.getQueryExecutor(tenantId);
   }
 
-  private PgPool getCachecPool(String tenantId) {
+  private PgPool getCachePool(String tenantId) {
```
Comment (wwelling) [inline mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java:719]: Thanks for finding the typo. It should however be `getCachedPool`.

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java [hunk 2]
```diff
@@ -172,13 +176,15 @@ public RecordDaoImpl(final PostgresClientFactory postgresClientFactory) {
       .offset(offset)
       .limit(limit)
       .getSQL(ParamType.INLINED);
-    return getCachecPool(tenantId).rxBegin()
+
+    return getCachePool(tenantId)
```
Comment (wwelling) [inline mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java:180]: This is an incomplete conversion. This is no longer running in a transaction. Following the examples here https://vertx.io/docs/vertx-pg-client/java/#_streaming, it should be:

```
    return getCachedPool(tenantId)
      .rxGetConnection()
      .flatMapPublisher(conn -> conn.rxBegin()
        .flatMapPublisher(tx -> conn.rxPrepare(sql)
          .flatMapPublisher(pq -> pq.createStream(1)
            .toFlowable()
            .map(this::toRow)
            .map(this::toRecord))
          .doAfterTerminate(tx::commit)));
```

This affords to fetch the cursor with a single row and stream the remaining.

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java [hunk 3]
```diff
@@ -443,13 +449,16 @@ public RecordDaoImpl(final PostgresClientFactory postgresClientFactory) {
...
```
Comment (wwelling) [inline mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java:453]: Same as above, no longer in a transaction. Should be:

```
    return getCachedPool(tenantId)
      .rxGetConnection()
      .flatMapPublisher(conn -> conn.rxBegin()
        .flatMapPublisher(tx -> conn.rxPrepare(sql)
          .flatMapPublisher(pq -> pq.createStream(1)
            .toFlowable()
            .map(this::toRow)
            .map(this::toSourceRecord))
          .doAfterTerminate(tx::commit)));
```

**Review by wwelling — CHANGES_REQUESTED** [review]
Good upgrade and cleanup. Two changes requested; typo correction should be to `getCachedPool` and to stream results in a transaction.

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java (hunk 1 above)
Comment (VRohach) [reply mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java]: @wwelling done

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java (hunk 2 above)
Comment (VRohach) [reply mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java]: @wwelling done

File: mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java (hunk 3 above)
Comment (VRohach) [reply mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java]: @wwelling done

**Comment by VRohach** [comment]
> Good upgrade and cleanup. Two changes requested; typo correction should be to `getCachedPool` and stream results should be in a transaction affording fetching single result and continue streaming with cursor.

@wwelling  Thank you for your comments! I changed that, could you please double-check?

**Review by wwelling — APPROVED** [review]
Approved pending RMB upgrade of mod-pubsub to fix test and passing SonarQube.

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -246,20 +246,30 @@
...
             "pubsub.subscribers.post"
           ]
+        },
+        {
+          "methods": [
+            "DELETE"
+          ],
+          "pathPattern": "/_/tenant/{id}",
+          "modulePermissions": [
+            "pubsub.event-types.post",
```
[the remark this answers is not in the document]
Comment (VRohach) [reply descriptors/ModuleDescriptor-template.json]: @KaterynaSenchenko  Thanks. Changed.

**Review by KaterynaSenchenko — APPROVED** [review]

**Review by RuslanLavrov — APPROVED** [review]

**Review by OleksiiKuzminov — APPROVED** [review]

**Pull request merged**